### PR TITLE
feat: Upgrade to openai-api-rs 2.0.0

### PR DESCRIPTION
**Pull request by jpmcb**
## Description

The types in openai-api-rs didn't change but this uses the `new` methods for the `ChatCompletionRequest` implementations / types

Once we get this in, should be able to cut a new release to `beta` and get this deployed.

Also ran `make lint` which caught a few linting errors

## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Related Tickets & Documents

Closes https://github.com/open-sauced/repo-query/pull/57

## Added tests?

- [ ] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [ ] 🙅 no documentation needed

### CODE REVIEW

**Review by Anush008 — APPROVED** [review]
🍕🚀🚀

File: src/conversation/prompts.rs [hunk 1]
```diff
@@ -17,23 +17,10 @@ pub fn generate_completion_request(
...
-        n: None,
-        stream: None,
-        stop: None,
-        max_tokens: None,
-        presence_penalty: None,
-        frequency_penalty: None,
-        logit_bias: None,
-        user: None,
-    }
+    ChatCompletionRequest::new(CHAT_COMPLETION_MODEL.to_string(), messages)
```
Comment (nickytonline) [inline src/conversation/prompts.rs:20]: I know it's related to the version update, but what's the TLDR; on the difference here?

Comment (Anush008) [reply src/conversation/prompts.rs:20]: The library moved to a builder pattern.
Before we had to do something like this.
```rust
let req = ChatCompletionRequest {
    model: chat_completion::GPT4.to_string(),
    messages: vec![chat_completion::ChatCompletionMessage {
        role: chat_completion::MessageRole::user,
        content: String::from("What is Bitcoin?"),
        name: None,
        function_call: None,
    }],
    functions: None,
    function_call: None,
    temperature: 0.7,
    top_p: None,
    n: None,
    stream: None,
    stop: None,
    max_tokens: None,
    presence_penalty: None,
    frequency_penalty: None,
    logit_bias: None,
    user: None,
};
```
Now it's
```rust
let req = ChatCompletionRequest::new(
    chat_completion::GPT4.to_string(),
    vec![chat_completion::ChatCompletionMessage {
        role: chat_completion::MessageRole::user,
        content: String::from("What is Bitcoin?"),
        name: None,
        function_call: None,
    }],
).temperature(0.7);
```
Making it a lot less verbose and concise.

**Review by nickytonline — APPROVED** [review]
🚀

File: src/conversation/prompts.rs (hunk 1 above)
Comment (jpmcb) [reply src/conversation/prompts.rs:20]: We could _technically_ still use the types directly. But that's sort of a pain and like Anush said, it'll be way more concise and easy to read / maintain in the future.

**Pull request merged**